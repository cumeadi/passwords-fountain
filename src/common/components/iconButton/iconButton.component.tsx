import { h } from 'preact';
import { TypedComponent } from '@/common/typings/prop-types';
import PropTypes from 'prop-types';
import { Wrapper } from './iconButton.styles';
import { Icon } from '@/common/components/icon';
import { IconName } from '@/assets/icons';

export const IconButton: TypedComponent<Props> = ({
    iconName,
    width,
    height,
}: Props) => {
    return (
        <Wrapper>
            <Icon name={iconName} width={width} height={height} />
        </Wrapper>
    );
};

interface Props {
    iconName: IconName;
    width?: number;
    height?: number;
}

IconButton.propTypes = {
    iconName: PropTypes.any.isRequired,
    width: PropTypes.number,
    height: PropTypes.number,
};

IconButton.defaultProps = {
    width: 30,
    height: 30,
};
