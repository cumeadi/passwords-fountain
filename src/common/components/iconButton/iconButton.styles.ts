import { h } from 'preact';
import { styled, setPragma } from 'goober';
import { theme } from '@/common/theme';
import { Wrapper as Button } from '@/common/components/button/button.styles';
setPragma(h);

export const Wrapper = styled(Button)`
    padding: ${theme.spacing.xs6};
    min-width: initial;
`;
