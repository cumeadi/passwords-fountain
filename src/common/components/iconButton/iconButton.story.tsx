import { h, VNode } from 'preact';
import { IconButton } from './iconButton.component';
import { withA11y } from '@storybook/addon-a11y';

export default {
    title: 'IconButton',
    decorators: [withA11y],
};

export const defaultView = (): VNode => (
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    <IconButton onClick={(): void => {}} iconName="bin" />
);
