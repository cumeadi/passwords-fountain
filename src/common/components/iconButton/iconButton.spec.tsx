import '@testing-library/jest-dom/extend-expect';
import { h } from 'preact';
import { render } from '@testing-library/preact';
import { IconButton } from './iconButton.component';

describe('IconButton', () => {
    it('should render correctly', () => {
        const { asFragment } = render(
            <IconButton
                iconName="bin"
                onClick={() => {
                    /* eslint-disable @typescript-eslint/no-empty-function */
                }}
            />
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
