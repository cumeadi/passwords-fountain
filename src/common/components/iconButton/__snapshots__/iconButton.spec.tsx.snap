// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should render correctly 1`] = `
<DocumentFragment>
  <button
    class="go3135434547 go3618827259"
  >
    <span
      class="go4062099715"
    >
      <img
        alt="icon"
        class="go115679"
        height="30"
        width="30"
      />
    </span>
  </button>
</DocumentFragment>
`;
